force-app/main/default/classes/Addresses.cls

```diff
@@ -165,210 +166,293 @@ public inherited sharing class Addresses extends fflib_SObjects {
+        Map<Id, List<Address__c>> allAddressesByAccountId = getAllAddressesByAccountId(accountIds);
 
         // for each hh, find the current address to use (seasonal vs default)
-        Map<Id, HHInfo> mapAccIdHHInfo = new Map<Id, HHInfo>();
-        for (Id accId : listAccId) {
-            // create a HHInfo for each household
-            HHInfo hhinfo = new HHInfo();
-            hhinfo.accId = accId;
-            mapAccIdHHInfo.put(accId, hhinfo);
+        Map<Id, HouseholdInfo> householdInfoByAccountId = getHouseholdAccountAddressInfoByAccountId(new
+                Set<Id>(accountIds),
+                defaultSeasonalAddressesByAccountId, allAddressesByAccountId);
 
-            // if there is a provided Default address, use it over all other defaults.
-            Address__c addr = mapAccIdAddr.get(accId);
-            if (addr != null && addr.Default_Address__c) {
-                hhinfo.addrDefault = addr;
-            }
+        // some callers just need the HHInfo for each HH Account.
+        if (shouldReturnHouseholdInfoOnly) {
+            return householdInfoByAccountId;
+        }
 
-            // go through the hh addresses storing the default and any seasonal
-            if (mapAccIdListAddrHH.get(accId) != null) {
-                for (Address__c addrHH : mapAccIdListAddrHH.get(accId)) {
-                    NPSP_Address npspAddressHH = new NPSP_Address(addrHH);
+        // get all Contacts by HH, that might need updating
+        // note that if we have an organizational account, our query will not pick up any contacts (by design!)
+        Map<Id, List<Contact>> contactsByHouseholdAccountId = getContactsByHouseholdAccountId(new Set<Id>(accountIds));
 
-                    if (hhinfo.addrDefault == null && addrHH.Default_Address__c) {
-                        hhinfo.addrDefault = addrHH;
-                    }
-                    if (npspAddressHH.isSeasonalAddr()) {
-                        Date dtStart = dtStartOrEndOfAddrSeasonal(addrHH, true);
-                        Date dtEnd = dtStartOrEndOfAddrSeasonal(addrHH, false);
-                        Date dtToday = System.today();
-                        if (dtToday >= dtStart && dtToday < dtEnd) {
-                            hhinfo.addrSeasonal = addrHH;
-                        }
-                    }
-                }
-            }
+        List<Account> accountsToUpdate = new List<Account>();
+        List<Address__c> addressesToUpdate = new List<Address__c>();
+        List<Contact> contactsToUpdate = new List<Contact>();
 
-            // finally, if no default found, create an empty default
-            if (hhinfo.addrDefault == null) {
-                Address__c addrEmpty = new Address__c();
-                addrEmpty.Household_Account__c = accId;
-                hhinfo.addrDefault = addrEmpty;
-            }
+        // now go through each HH, and do all the work!
+        updateAddressInfoForAccountsAndContacts(
```

Comment:
One possible idea for shortening the signature of this method, can we have it return an instance of a class that contains `accountsToUpdate`, `addresseesToUpdate` and `contactsToUpdate`?

Reply:
Something like `AddressAndRelationships`

Reply:
great recommendation!